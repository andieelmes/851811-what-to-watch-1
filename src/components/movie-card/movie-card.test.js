import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import MovieCard from 'App/components/movie-card/movie-card';

const movie = {
  id: 1,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
};

describe(`Movie card component`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(<BrowserRouter>
        <MovieCard
          {...movie}
          onHover={() => {}}
        /></BrowserRouter>, {createNodeMock: (el) => el})
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
