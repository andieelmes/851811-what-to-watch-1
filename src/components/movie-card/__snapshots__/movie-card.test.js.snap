// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card component should render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-movie-id={1}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="small-movie-card__image"
    href="/film/1"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      style={null}
      width="280"
    />
    <video
      controls={true}
      loop={true}
      muted={true}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film/1"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
